README.md

```diff
@@ -58,42 +95,235 @@ import ReactDOM from 'react-dom';
 import stores from './stores';
 
 function Todo() {
-  const todos = stores.useStore('todos');
-  const { dataSource } = todos;
+  const todos = stores.useStore("todos");
+  const { dataSource, refresh, add, remove, toggle } = todos;
 
-  useEffect(() => {
-    todos.refresh();
+  React.useEffect(() => {
+    refresh();
   }, []);
 
+  function onAdd(name) {
+    add({ name });
+  }
+
+  function onRemove(index) {
+    remove(index);
+  }
+
+  function onCheck(index) {
+    toggle(index);
+  }
+
   return (
     <div>
-      <h2>Todo</h2>
-      <ul>
-        {dataSource.map(({ name }) => (
-          <li>{name}</li>
-        ))}
-      </ul>
+      <h2>Todos</h2>
+      {!refresh.loading ? (
+        dataSource.length ? (
+          <ul>
+            {dataSource.map(({ name, done }, index) => (
+              <li key={index}>
+                <label>
+                  <input
+                    type="checkbox"
+                    checked={done}
+                    onClick={() => onCheck(index)}
+                  />
+                  {done ? <s>{name}</s> : <span>{name}</span>}
+                </label>
+                <button onClick={() => onRemove(index)}>-</button>
+              </li>
+            ))}
+          </ul>
+        ) : (
+          "no task"
+        )
+      ) : (
+        "loading..."
+      )}
+      <div>
+        <input
+          onKeyDown={event => {
+            if (event.keyCode === 13) {
+              onAdd(event.target.value);
+              event.target.value = "";
+            }
+          }}
+          placeholder="Press Enter"
+        />
+      </div>
     </div>
   );
-};
+}
+
+const rootElement = document.getElementById("root");
+ReactDOM.render(<Todo />, rootElement);
+```
+
+Complete example is presented in this [sandbox](https://codesandbox.io/s/icestore-hs9fe), feel free to play with it.
+
+## API
+
+### registerStore
+
+Register store config to global store instance. 
+
+* Paramters
+  - namespace {string} unique name of store
+  - bindings {object} object of store config including state and actions
+* Return value
+  - {object} store instance
+
+### useStores
+
+Hooks of using multiple store.
+
+* Paramters
+  - namespaces {array} arrays of store namespace
+* Return value
+  - {array} array of stores' bindings
+
+### useStore
+
+Hooks of using single store.
+
+* Paramters
+  - namespace {string} store namespace
+* Return value
+  - {object} store's bindings
 
-ReactDOM.render(
-  <Todo />,
-  document.body
+## Advanced use
+
+### async status
+
+icestore has built in async status support in async function, it enables user to access to async functions' loading and error status without defining extra state which will make code more consise and clean.
```

Comment:
icestore has built-in async support for actions. This enables users to have access to loading and error status asynchronously without defining an extra state, making the code more concise and clean.

Reply:
> icestore has built-in async support for actions. This enables users to have access to loading and error status asynchronously without defining an extra state, making the code more concise and clean.

To clarify, `async status` here means the executing status of async actions, not async actions itself, so i tweak your above suggestions a little bit to avoid confusing:

###  async actions' executing status

icestore has built-in executing status support for async actions. This enables users to have access to the loading and error executing status of async actions without defining extra state, making the code more consise and clean.

Please check the above modified version again.

Reply:
I see! For the first sentence, I think this sounds better:

```
icestore has built-in support to access the executing status of async actions.
```

Reply:
OK, got it.